services/ui/ws/display.cc

```diff
@@ -361,6 +361,22 @@ void Display::OnCloseRequest() {
     window_tree->client()->RequestClose(window_id.id);
 }
 
+// TODO(msisov, tonikitoo): maybe combine calls to get window_tree,
+// server_window and window_id to duplicating the same code used above?
+void Display::OnWindowStateChanged(ui::mojom::ShowState new_state) {
+  if (!window_server_->IsInExternalWindowMode())
+    return;
+
+  WindowTree* window_tree = window_server_->GetTreeForExternalWindowMode();
+  WindowManagerDisplayRoot* display_root =
```

Comment:
I would add a comment like this:

// TODO(msisov,tonikitoo): The code below should be in window_tree.cc. I added it here for the sake of adding less
// code as possible downstream.